kythe/go/extractors/config/runextractor/cmakecmd/cmakecmd.go

```diff
@@ -107,7 +107,9 @@ func (c *cmakeCommand) Execute(ctx context.Context, fs *flag.FlagSet, args ...in
 		return c.Fail("Unable to create build directory: %v", err)
 	}
 
-	if err := runIn(exec.CommandContext(ctx, "cmake", "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON", sourceDir), buildDir); err != nil {
+	// The Kythe extractor is clang based, so tell cmake that we are Clang so the right
+	// commands are generated.
+	if err := runIn(exec.CommandContext(ctx, "cmake", "-DCMAKE_CXX_COMPILER_ID=Clang", "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON", sourceDir), buildDir); err != nil {
```

Comment:
Does this actually effect out CMake *picks* a compiler or just which compiler it asserts is in use?   I couldn't find it documented, but I worry that if it's just the latter the project may fail to compile due to a mismatch.

Reply:
switched to tell cmake to actually use clang rather than just report our ID as clang. This should be safer and we can only support builds that would normally work with clang anyway, so assuming they have a clang binary installed seems ok.